torch_geometric/nn/encoding.py

```diff
@@ -94,3 +97,178 @@ def forward(self, x: Tensor) -> Tensor:
+        h = torch.nn.functional.dropout(h, p=0.5, training=self.training)
+        h_channel = h + h_token
+
+        # head
+        h_channel = self.head_layer_norm(h_channel)
+        t = torch.mean(h_channel, dim=1)
+        return self.head_lin(t)
+
+    def __repr__(self) -> str:
+        return (f"{self.__class__.__name__}("
+                f"num_tokens={self.num_tokens}, "
+                f"in_channels={self.in_channels}, "
+                f"out_channels={self.out_channels})")
+
+
+class LinkEncoding(torch.nn.Module):
+    r"""The link-encoding function from the `"Do We Really Need Complicated
+    Model Architectures for Temporal Networks?"
+    <https://openreview.net/forum?id=ayPPc0SyLv1>`_ paper.
+    :class:`LinkEncoding` is composed of two components. The first component is
+    :class:`TemporalEncoding` that maps each edge timestamp to a
+    ``time_channels`` dimensional vector.
+    The second component, 1-layer MLP-mixer, maps each encoded timestamp
+    feature concatenated with its corresponding link feature to a
+    ``out_channels`` dimensional vector.
+
+    Args:
+        K (int): The number of most recent teomporal links to use to construct
+            an intermediate feature representation for each node.
+        num_edge_features (int): The number of edge features used to construct
+            a linear layer encoding the output of :class:`TemporalEncoding`
+            concatenated with ``edge_attr``.
+        time_channels (int): dims to encode each timestamp into with
+            :class:`TemporalEncoding`.
+        out_channels (int): Size of each output sample.
+
+    Example:
+
+        >>> # GraphMixer paper uses the following args for GDELTLite dataset
+        >>> link_encoder = LinkEncoding(
+        ...     K=30,
+        ...     num_edge_features=186,
+        ...     hidden_channels=100,
+        ...     out_channels=100,
+        ...     time_channels=100,
+        ... )
+
+    """
+    def __init__(
+        self,
+        K: int,
+        num_edge_features: int,
+        hidden_channels: int,
+        out_channels: int,
+        time_channels: int,
+    ) -> None:
+        super().__init__()
+        self.K = K
+        self.num_edge_features = num_edge_features
+        self.hidden_channels = hidden_channels
+        self.out_channels = out_channels
+        self.time_channels = time_channels
+
+        # teomporal encoder
+        self.temporal_encoder = TemporalEncoding(time_channels)
+        self.temporal_encoder_head = torch.nn.Linear(
+            time_channels + num_edge_features,
+            hidden_channels,
+        )
+
+        # temporal information summariser
+        self.mlp_mixer = _MLPMixer(
+            num_tokens=K,
+            in_channels=hidden_channels,
+            out_channels=out_channels,
+        )
+
+    def forward(
+        self,
+        edge_attr: Tensor,
+        edge_time: Tensor,
+        node_batch: Tensor,
+        num_nodes: Optional[int] = None,
```

Comment:
I don't think we need these two arguments, see above comment.
But we should add an argument `is_sorted` to indicate if the `edge_index` is sorted by `dst` node.